src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,30 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login should be at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Password should be at least 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new RegistrationException("Age should be at least 18 years old");
```

Comment:
Don't create one big if with all conditions. Better to split one big if into several small ones. This allows for more informative exception messages, indicating exactly which input was invalid.

Reply:
Done